test/lib/mayaUsd/render/mayaToHydra/cpp/testIsolateSelectSwitchToVP2.py

```diff
@@ -0,0 +1,163 @@
+
+    _pluginsToLoad = ['mayaHydraCppTests']
+    _pluginsToUnload = []
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestIsolateSelectSwitchToVP2, cls).setUpClass()
+        for p in cls._pluginsToLoad:
+            if not cmds.pluginInfo(p, q=True, loaded=True):
+                cls._pluginsToUnload.append(p)
+                cmds.loadPlugin(p, quiet=True)
+
+    @classmethod
+    def tearDownClass(cls):
+        super(TestIsolateSelectSwitchToVP2, cls).tearDownClass()
+        # Clean out the scene to allow all plugins to unload cleanly.
+        cmds.file(new=True, force=True)
+        for p in reversed(cls._pluginsToUnload):
+            cmds.unloadPlugin(p)
+
+    def setupScene(self):
+
+        cmds.polyTorus()
+        cmds.polySphere()
+        cmds.polyCube()
+
+        cmds.refresh()
+
+        scene = [
+            # Maya objects
+            '|pTorus1', 
+            '|pTorus1|pTorusShape1', 
+            '|pSphere1',
+            '|pSphere1|pSphereShape1',
+            '|pCube1',
+            '|pCube1|pCubeShape1']
+
+        return scene
+
+    def assertVisible(self, visible):
+        for v in visible:
+            self.trace("Testing %s for visibility\n" % v)
+            cmds.mayaHydraCppTest(v, f="TestHydraPrim.isVisible")
+
+    def assertNotVisible(self, notVisible):
+        for nv in notVisible:
+            self.trace("Testing %s for invisibility\n" % nv)
+            cmds.mayaHydraCppTest(nv, f="TestHydraPrim.notVisible")
+
+    def assertVisibility(self, visible, notVisible):
+        self.assertVisible(visible)
+        self.assertNotVisible(notVisible)
+
+    def test_isolateSelectSwitchToVP2(self):
+
+        scene = self.setupScene()
+        modelPanel = 'modelPanel4'
+        enableIsolateSelect(modelPanel)
+ 
+        # Isolate select Maya object.
+        cmds.select('|pTorus1')
+        cmds.editor(modelPanel, edit=True, updateMainConnection=True)
+        cmds.isolateSelect(modelPanel, loadSelected=True)
+
+        visible = ['|pTorus1', '|pTorus1|pTorusShape1']
+
+        notVisible = scene.copy()
+        for p in visible:
+            notVisible.remove(p)
+
+        cmds.refresh()
+
+        self.assertVisibility(visible, notVisible)
+
+        # Record the number of view selected changed calls so far.
+        viewSelectedChangedCallsPre = cmds.mayaHydraInstruments("MtohRenderOverride:NbViewSelectedChangedCalls", q=True)
+
+        self.assertGreater(viewSelectedChangedCallsPre, 0)
+
+        # Switch to VP2, and make a different isolate selection.  The number of
+        # Hydra view selected changed calls should remain the same.
+        self.setViewport2Renderer()
+
+        cmds.select('|pSphere1')
+        cmds.editor(modelPanel, edit=True, updateMainConnection=True)
+        cmds.isolateSelect(modelPanel, loadSelected=True)
+
+        cmds.refresh()
+
+        viewSelectedChangedCallsVP2 = cmds.mayaHydraInstruments("MtohRenderOverride:NbViewSelectedChangedCalls", q=True)
```

Comment:
When switching to VP2 and performing isolate select, the mayaHydra callback should early out.